docs/HOL_BDD-Security-Framework-integration-into-VSTS/README.md

```diff
@@ -0,0 +1,317 @@
+
+* Test Result Format: `JUnit`
+* Test Results Files: `$(System.DefaultWorkingDirectory)/all_tests.xml`
+* Test Run Title: `Security Test`
+* Always run: Ticked
+
+    ![](media/21.png)
+
+**Step 9.** Save the changes by clicking on "Save" and then "OK".
+
+  ![](media/22.png)
+
+**Step 10.** Create a new release to run the security tests. Click on "Release" button, select "Create Release", select the latest build(created at the beginning of this task) and click on "Create".
+
+  ![](media/23.png)
+
+**Step 11.** Once the release has successfully completed, lets check test result.
+
+1. Hover over "Test" and then click on "Runs".
+
+    ![](media/27.png)
+
+2. Double click on the latest test results.
+
+    ![](media/28.png)
+
+3. All of the 16 selected tests should have passed and you should see the following summary.
+
+    ![](media/29.png)
+
+
+
+### Task 4: Update BDD-Security to include more tests
+
+**Step 1.** Lets add more tests to be run by BDD-Security. Navigate to the location where you keep your PartsUnlimitedMRP repository and open **SSH-MRP-BDD-Security.ps1** script located at **PartsUnlimitedMRP -> deploy**.
+
+
+**Step 2.** Replace `./gradlew -Dcucumber.options="--tags @app_scan --tags ~@skip"` with:
+
+ ```
+ ./gradlew -Dcucumber.options="--tags @app_scan,@host_config --tags ~@skip"
+ ```
+
+
+**Step 3.** At this stage you should also address a possibility of insecure website being promoted to production environment. Add the following code at the end of the script to fail the release if there were errors in test results:
+```
+# If file with results contains "failed" then returning non-zero value will indicate error and fail the build.
+return @(Get-Content all_tests.xml | Where-Object { $_.Contains("failed") }).Count
+```
+>**Note:** This is not required to fail the build if you have "Fail on Standard Error" ticked (by default) on "BDD-Security" task.
+    ![](media/30.png)
+
+
+**Step 4.** Save `SSH-MRP-BDD-Security.ps1`. Open command line (which supports git) and navigate to your PartsUnlimitedMRP repository location. Run the following commands to add these changes to the remote repo and trigger automatic build and release.
+
+```
+git add .
+git commit -m "added more tests to run by SSH-MRP-BDD-Security.ps1"
+git push
+```
+
+
+**Step 5.** Once the build and release have completed executing, open test runs in VSTS again and click on the latest result.
+
+  ![](media/31.png)
+
+
+**Step 6.** You should see the summary indicating that one test has failed. To investigate the reason of failure, download the attachment `all_tests.xml` by clicking on it.
+
+  ![](media/32.png)
+
+
+**Step 7.** Open downloaded `all_tests.xml` in text editor and find which test has failed. Based on the stack trace, a port test has failed by finding open ports other than 80 and 443. To fix this issue, you will need to modify the test to expect 22 to be open and run tests again.
+
+  ![](media/33.png)
+
```

Comment:
For the Step 6 and 7, I would prefer to do not download the file and check the failed test in the tests results section.